Core/TransactionHandler.php

```diff
@@ -0,0 +1,305 @@
+        $oTransaction->setParentTransactionId($sParentTransactionId);
+
+        if (!is_null($fAmount)) {
+            $sCurrencyName = $oOrder->oxorder__oxcurrency->value;
+            $oCurrency = Registry::getConfig()->getCurrencyObject($sCurrencyName);
+
+            $oTransaction->setAmount(new Amount(
+                Registry::getUtils()->fRound($fAmount, $oCurrency),
+                $sCurrencyName
+            ));
+        }
+
+        $oBackendService = new BackendService($oConfig, $this->_oLogger);
+        $oResponse = $oBackendService->process($oTransaction, $sActionTitle);
+
+        return $this->_onActionResponse($oResponse, $oOrder, $oBackendService);
+    }
+
+    /**
+     * Handles the success and error response from the post-processing action.
+     *
+     * @param Response       $oResponse
+     * @param Order          $oOrder
+     * @param BackendService $oBackendService
+     *
+     * @return array either [status => success] or [status => error, message => errorMessage]
+     *
+     * @since 1.0.1
+     */
+    private function _onActionResponse($oResponse, $oOrder, $oBackendService)
+    {
+        if ($oResponse instanceof FailureResponse) {
+            return $this->_onActionFailure($oResponse);
+        }
+
+        if ($oResponse instanceof SuccessResponse) {
+            return $this->_onActionSuccess($oResponse, $oBackendService);
+        }
+
+        return $this->_getErrorMessage('No handler for this response type implemented');
+    }
+
+    /**
+     * Handle the success response case.
+     *
+     * @param SuccessResponse $oResponse
+     * @param BackendService  $oBackendService
+     *
+     * @return array
+     *
+     * @since 1.0.1
+     */
+    private function _onActionSuccess($oResponse, $oBackendService)
+    {
+        $sRootTransactionId = $this->_getRootTransactionId($oResponse->getParentTransactionId());
+        $oOrder = oxNew(Order::class);
+        if (!$oOrder->loadWithTransactionId($sRootTransactionId)) {
+            $sErrorMessage = 'No order found for transactionId: ' . $oResponse->getParentTransactionId();
+            $this->_oLogger->error($sErrorMessage);
+            return $this->_getErrorMessage(Helper::translate('text_generic_error') . '<br>' . $sErrorMessage);
+        }
+
+        // the reponse handler creates the transaction entry in the database
+        ResponseHandler::onSuccessResponse($oResponse, $oBackendService, $oOrder);
+        $this->_updateParentTransactionStateIfNecessary($oResponse);
+        OrderHelper::updateOrderState($oOrder, $oResponse, $oBackendService);
+
+        return $this->_getSuccessMessage();
+    }
+
+    /**
+     * Handles the failure response case.
+     *
+     * @param FailureResponse $oResponse
+     *
+     * @return array
+     *
+     * @since 1.0.1
+     */
+    private function _onActionFailure($oResponse)
+    {
+        $sErrorDescriptions = $this->_getErrorString($oResponse);
```

Comment:
Please change to singular variable name:

> sErrorDescription

Reply:
👍